.github/workflows/test.yml

```
@@ -735,3 +720,97 @@ jobs:
+        run: |
+          curl -fsSL https://get.buildpulse.io/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
+          chmod +x ./buildpulse-test-reporter
+          ./buildpulse-test-reporter submit packages --account-id 17219288 --repository-id 192925833
+
+  #
+  # Run Client tests on macOS and Windows. (Rasoning see above)
+  #
+  no-docker-client:
+    timeout-minutes: 40
+    runs-on: ${{ matrix.os }}
+    needs: detect_jobs_to_run
+
+    if: |
+      ${{
+        contains(needs.detect_jobs_to_run.outputs.jobs, '-client-')
+      }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-11, windows-latest]
+        node: [16]
+        queryEngine: ['library', 'binary']
+
+    env:
+      JEST_JUNIT_SUITE_NAME: '${{ matrix.os }}/node-${{ matrix.node }}/${{ matrix.queryEngine }}/{title}'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Prerequisites
+        shell: bash
+        run: |
+          echo "TEST_SKIP_MSSQL=true" >> $GITHUB_ENV
+          echo "TEST_SKIP_MONGODB=true" >> $GITHUB_ENV
+          echo "TEST_SKIP_COCKROACHDB=true" >> $GITHUB_ENV
+          echo "TEST_MONGO_URI_MIGRATE=mongodb://localhost:27017/tests-migrate" >> $GITHUB_ENV
+          echo "PRISMA_CLI_QUERY_ENGINE_TYPE=${{ matrix.queryEngine }}" >> $GITHUB_ENV
+          echo "PRISMA_CLIENT_ENGINE_TYPE=${{ matrix.queryEngine }}" >> $GITHUB_ENV
+
+      - name: Setup PostgreSQL
+        run: bash .github/workflows/setup-postgres.sh
+
+      - name: Setup MySQL
+        run: bash .github/workflows/setup-mysql.sh
+
+      - uses: pnpm/action-setup@v2.1.0
+        with:
+          version: latest
+
+      - uses: actions/setup-node@v2
+        with:
+          cache: 'pnpm'
+          node-version: ${{ matrix.node }}
+
+      - run: bash .github/workflows/setup.sh
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - name: Test packages/client
+        if: ${{ contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') || contains(needs.detect_jobs_to_run.outputs.jobs, '-client-') }}
+        run: pnpm run test-notypes
+        working-directory: packages/client
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+          NODE_OPTIONS: '--max-old-space-size=8096'
+
+      - uses: codecov/codecov-action@v2
+        if: ${{ contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') || contains(needs.detect_jobs_to_run.outputs.jobs, '-client-') }}
+        with:
+          files: ./packages/client/src/__tests__/coverage/clover.xml
+          flags: client,${{ matrix.os }},${{ matrix.queryEngine }}
+          name: client-${{ matrix.os }}-${{ matrix.queryEngine }}
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: ${{ !cancelled() }} # Run this step even when the tests fail. Skip if the workflow is cancelled.
```

👋 @janpio: In our email chat, I mentioned that we should only run this step on branches that are in this repository (as opposed to running it in forks, since forks don't have access to secrets). To set this up, we can use use this check (similar to the [approach used in Homebrew/brew](https://github.com/Homebrew/brew/blob/01506f120abd0d258afeb942b56deb3dd82470a9/.github/workflows/tests.yml#L264)):

```suggestion
        # Only run this step for branches where where we have access to secrets.
        # Run this step even when the tests fail. Skip if the workflow is cancelled.
        if: startsWith(github.repository, 'prisma/') && !cancelled()
```